logparser/logger.py

```diff
@@ -0,0 +1,304 @@
+# Log Parser for RTI Connext.
+#
+#   Copyright 2016 Real-Time Innovations, Inc.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Logger.
+
+The module contains the class to log the messages.
+"""
+
+
+class Logger(object):
+    """Class to log the messages.
+
+    Attributes:
+        COLORS: colors to use in the logs
+        KIND_TO_COLOR: logs messages to color
+        state (:obj:`list` of `str`): information about the parse process
+        verbosity (int): verbosity level of the log
+        inline (bool): show warnings/erros in network logs
+        ignorePackets (bool): ignore network events
+        ignore
+        showColors (bool): show colors in the log
+        formatDevice (:obj:`FormatDevice`): format device to print the logs
+        highlight (:obj:`compiled re`): show in bold regex matched logs
+        onlyIf (:obj:`compiled re`): show only
+            log messages that match the regex
+    """
+
+    def __init__(self, state):
+        """Constructor of the class."""
+        self._COLORS = {
+            'RED': '\033[91m',
+            'GREEN': '\033[92m',
+            'YELLOW': '\033[93m',
+            'BLUE': '\033[94m',
+            'MAGENTA': '\033[95m',
+            'BOLD': '\033[1m',
+            'FAINT': '\033[2m',
+            'ITALIC': '\033[3m',
+            'UNDERLINE': '\033[4m',
+            'END': '\033[0m',
+        }
+
+        self._KIND_TO_COLOR = {
+            'WARNING': 'YELLOW|ITALIC',
+            'ERROR': 'RED|BOLD',
+            'IMPORTANT': 'BOLD'
+        }
+
+        self._state = state
+        self._verbosity = 0
+        self._inline = True
+        self._ignorePackets = False
+        self._showColors = False
+        self._formatDevice = self._state['format_device']
+        self._highlight = None
+        self._onlyIf = None
+
+    def setVerbosity(self, value):
```

Comment:
I think that this is not a good pattern to follow... We are using classes and one of their objectives is "encapsulation". We don't need these attributes as public... What about if we want to change something in the future? If we use the attributes directly, we will need to change a lot of lines of code...
Also, I think it is more readable.

What about properties? This is the "pythonic" way